buildpacks/python/Makefile

```diff
@@ -45,10 +45,12 @@ buildpacks.python.invoker.sha := $(buildpacks.python.temp_dir)/invoker.sha256
 buildpacks.python.invoker_deps.url := $(base_url)/releases/download/python-invoker/v$(buildpacks.python.invoker.version)/pyfunc-invoker-deps-$(buildpacks.python.invoker.version).tar.gz
 buildpacks.python.invoker_deps.sha := $(buildpacks.python.temp_dir)/invoker_dep.sha256
 
+.PHONY: $(buildpacks.python.invoker.sha)  # ensure invoker is always fresh
```

Comment:
Adding these phonys was the missing piece (here and in the java makefile) to enable the following workflow:
1. Run `make integration-tests`, watch it pass
2. Make a breaking invoker code change
3. Run `make integration-tests`, watch it fail

Without this, step 3 doesn't work as planned and will be a false positive. This is because this task copies the invoker's sha file into the buildpack's own temp location. Without the phony, if this file is already present in the buildpack's temp area, it will skip this target and not grab the latest sha, even if the invoker was recently changed and built.

Reply:
I considered an alternative where we simply store this sha file ONLY in the invoker's build/out area instead of trying to copy/curl it to the buildpack's temp area -- that way changes due to a rebuilt invoker would be picked up automatically. This will work fine for local builds, but I didn't want to introduce a risk to the release process, which relies on this task to curl the remote sha from the github release. Having this task be always performed is less risky and introduces a minimum amount of redundancy (namely that the curl will always happen when rebuilding the buildpack, instead of only when the invoker changes).

Another alternative is to remove the phony and simply force a delete (in local build case only) of the sha files that were last curled into the buildpack's temp area (in addition to the `values.yaml` file next to them). This is definitely doable but slightly less straightforward. It has the advantage that it can run only for local builds. That said, there's no reason for this optimization as the only benefit will be in the remote case -- and since CI does the releasing with a clean slate, there would be no material impact.

Reply:
TL;DR ensuring that the sha files are copied over always -- even if the content hasn't actually changed -- is the best & simplest way to keep them in sync with the actual local invoker